### ISSUE TITLE
Illegal Castle by White Allowed!

### ISSUE THREAD

**Issue by pychess** [pychess/pychess#186]
Original [issue 186](https://code.google.com/p/pychess/issues/detail?id=186) reported by [joseph.north](https://code.google.com/u/joseph.north/) 2007-08-05

**What steps will reproduce the problem?**
1. See attachment.
**2.**
**3.**

**What is the expected output? What do you see instead?**
It should NOT have been allowed!  It was!!

What version of the product are you using? Latest.

Did you use an installed version of PyChess or did you run in from a
tarball/svn? Installed.

**Please provide any additional information below.**

**Comment by pychess** [comment]
- attached [ill1.pgn](https://code.google.com/p/pychess/issues/detail?id=186#c0)

**Comment by pychess** [comment]
Original [comment 1.](https://code.google.com/p/pychess/issues/detail?id=186#c1) posted by [berado88](https://code.google.com/u/108250154226472549192/) 2007-08-06

White 0-0-0 _is_ a legal move here by chess rules.
http://en.wikipedia.org/wiki/Rules\_of\_chess#Movement

**Comment by pychess** [comment]
Original [comment 2.](https://code.google.com/p/pychess/issues/detail?id=186#c2) posted by [lobais](https://code.google.com/u/lobais/) 2007-08-13
- changed status to: Invalid

**Comment by pychess** [comment]
Original [comment 3.](https://code.google.com/p/pychess/issues/detail?id=186#c3) posted by [joseph.north](https://code.google.com/u/joseph.north/) 2007-08-13

How could that have been a legal move, pleasw?!

**Comment by pychess** [comment]
Original [comment 4.](https://code.google.com/p/pychess/issues/detail?id=186#c4) posted by [lobais](https://code.google.com/u/lobais/) 2007-08-13

http://en.wikipedia.org/wiki/Castling
Castling is a special move in the game of chess involving the king and either of the
original rooks of the same color. Castling consists of moving the king two squares
towards a rook, then moving the rook onto the square over which the king crossed.

**Comment by pychess** [comment]
Original [comment 5.](https://code.google.com/p/pychess/issues/detail?id=186#c5) posted by [joseph.north](https://code.google.com/u/joseph.north/) 2007-08-13

Dear "lobais":

   Castling is a fairly complicated Chess move, subject to more conditions
than you have elaborated, I believe!
   The move I described should NOT have been allowed, I continue to
believe!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

```
                      Sincerement,


                      Joseph Roy D. North
                      le     Roi Du Nord
```

Scientia est Potentia!

**Comment by pychess** [comment]
Original [comment 6.](https://code.google.com/p/pychess/issues/detail?id=186#c6) posted by [lobais](https://code.google.com/u/lobais/) 2007-08-14

Dear Joseph Roy D. North.
I feel very sorry if I have mistreated you in any way.

I have looked carefully through the pgn you provided and I really haven't been able
to pinpoint the problem you describe.
I can see that the white rook moves over an attacked square, which should be fine,
and I can see black castle after it has been check (note: Not _while_ it is check)
which should be fine as well.

_Please_ add information of the chess rule that PyChess doesn't seem to understand,
as well as the color who performed it.

Thomas D. Ahle
